### PR TITLE
Make encryption-related connectivity tests more reliable

### PR DESCRIPTION
**Pull request by giorio94**
Please refer to the individual commit messages for additional details.

### CODE REVIEW

**Comment by giorio94** [comment]
Closed and reopened to retrigger tests.

**Comment by giorio94** [comment]
EKS (tunnel) failure is addressed by https://github.com/cilium/cilium-cli/pull/2045
Marking ready for review again

**Comment by giorio94** [comment]
Converting back to draft while investigating one failure surfaced in https://github.com/cilium/cilium/actions/runs/6544264527/job/17770427183

**Review by nebril — APPROVED** [review]

**Comment by giorio94** [comment]
Back in review state

File: connectivity/tests/encryption.go [hunk 1]
```diff
@@ -56,14 +56,16 @@ func getInterNodeIface(ctx context.Context, t *check.Test, clientHost *check.Pod
 	return device
 }
 
-// getSourceAddress determines the source IP address we want to use for
-// capturing packet. If direct routing is used, the source IP is the client IP.
-func getSourceAddress(ctx context.Context, t *check.Test, client,
+// getSourceAddressFilter constructs the source IP address filter we want to use for
+// capturing packet. If direct routing is used, the source IP is the client IP,
+// otherwise, either the client IP or the one it can be masqueraded to.
```
Comment (brb) [inline connectivity/tests/encryption.go:78]: Nit: I think in the node-to-node encryption w/ tunneling on case the IP addr is just cilium_host IP addr, and it is not masqueraded to.

**Review by brb — APPROVED** [review]
Nice! Small nit.

File: connectivity/tests/encryption.go (hunk 1 above)
Comment (brb) [reply connectivity/tests/encryption.go:78]: I'm referring to the following comment part:

> or the one it can be masqueraded to

Comment (giorio94) [reply connectivity/tests/encryption.go:78]: Ah OK, with that I was referring to the output of `ip -o route get 10.0.0.1 | grep -oE 'src [^ ]*' | cut -d' ' -f2` (which had been added in https://github.com/cilium/cilium-cli/pull/1870).

Comment (brb) [reply connectivity/tests/encryption.go:78]: I just wanted to say that no masquerading happening in that case, and cilium_host's IP addr is chosen by the routing layer.

Comment (giorio94) [reply connectivity/tests/encryption.go]: I've rephrased the comment, and removed the reference to the masquerade process.

**Comment by brb** [comment]
The EKS failure looks legit - https://github.com/cilium/cilium-cli/actions/runs/6550230402/job/17788873914?pr=2035.

**Comment by giorio94** [comment]
> The EKS failure looks legit - https://github.com/cilium/cilium-cli/actions/runs/6550230402/job/17788873914?pr=2035.

Yes, it is addressed by https://github.com/cilium/cilium-cli/pull/2045 (that's why the PR is still blocked). Essentially node to pod connectivity is broken due to the leftover iptables chains, and apparently we are not testing that path anywhere else.

**Comment by giorio94** [comment]
Kind / installation-and-connectivity (helm) hit https://github.com/cilium/cilium-cli/issues/1894

**Pull request merged**